SDKBuildScripts/unity_buildAppCenterTestIOS.sh

```diff
@@ -24,94 +24,116 @@ ProjectFolderName=$(basename "$XCodeWorkspaceDirectory")
 GitRepoFolderName=$(basename "$AppCenterGitRepoURL" | sed -e 's/.git//g')
 
 #remove cruft from previous runs, if any.
-rm -fdr "$RepoWorkingDirectory/$GitRepoFolderName"
-mkdir -p "$RepoWorkingDirectory/$GitRepoFolderName"
-
-#create the appcenter prebuild script in the xcode project
-pushd "$XCodeWorkspaceDirectory"
-touch appcenter-post-clone.sh
-echo '#!/usr/bin/env bash
-
-touch Gemfile
-
-echo -e "source \"https://rubygems.org\"\n\ngem \"calabash-cucumber\", \">= 0.16\", \"< 2.0\"" > Gemfile
-
-bundle ;
-bundle exec calabash-ios download' > appcenter-post-clone.sh
-popd
-
-pushd "$RepoWorkingDirectory"
-
-#clone the appcenter build repo to our local workspace
-NewBranch=0
-git clone "$AppCenterGitRepoURL"
-cd $(basename "$AppCenterGitRepoURL" | sed -e 's/.git//g')
-git fetch --tags 
-git reset --hard $AppCenterGitRepoCleanTag
-git push --force
-
-git checkout "$AppCenterGitRepoBranchName"
-if [ $? -ne 0 ] 
-then
-    echo "Failed to checkout existing branch: $AppCenterGitRepoBranchName. Creating as new branch."
-    git checkout -b "$AppCenterGitRepoBranchName"
-    NewBranch=1
-fi
-
-#copy the xcode workspace into the appcenter git repo
-cp -r "$XCodeWorkspaceDirectory" "$RepoWorkingDirectory/$GitRepoFolderName"
-git add .
-git update-index --chmod=+x "$RepoWorkingDirectory/$GitRepoFolderName/$ProjectFolderName/MapFileParser.sh"
-git update-index --chmod=+x "$RepoWorkingDirectory/$GitRepoFolderName/$ProjectFolderName/appcenter-post-clone.sh"
-git commit -m "add xcode project for appcenter build"
-
-#if a new branch was created AppCenter needs to be manually configured for this branch.  
-if [ $NewBranch -eq 1 ]; then
-git push -u origin "$AppCenterGitRepoBranchName"
-echo 'ERROR: Unity Job '"$AppCenterGitRepoBranchName"' did not yet exist.'
-echo "The branch has been created and populated, but must be manually configured in AppCenter."
-echo "Please try again once the AppCenter build branch has been properly configured."
-false
-exit
-fi
-
-git push
+ResetRepositories() {
+    rm -fdr "$RepoWorkingDirectory/$GitRepoFolderName"
+    mkdir -p "$RepoWorkingDirectory/$GitRepoFolderName"
+
+    #create the appcenter prebuild script in the xcode project
+    pushd "$XCodeWorkspaceDirectory"
+    touch appcenter-post-clone.sh
+    echo '#!/usr/bin/env bash
+
+    touch Gemfile
+
+    echo -e "source \"https://rubygems.org\"\n\ngem \"calabash-cucumber\", \">= 0.16\", \"< 2.0\"" > Gemfile
+
+    bundle ;
+    bundle exec calabash-ios download' > appcenter-post-clone.sh
+    popd
+
+    pushd "$RepoWorkingDirectory"
+
+    #clone the appcenter build repo to our local workspace
+    NewBranch=0
+    git clone "$AppCenterGitRepoURL"
+    cd $(basename "$AppCenterGitRepoURL" | sed -e 's/.git//g')
+    git fetch --tags 
+    git reset --hard $AppCenterGitRepoCleanTag
```

Comment:
I blobbed it all into a single function because it was all git logic, and was resetting, and pushing clean copies of branches and stuff.
I'd be happy to break it down further, but I was more focused on the fixes I made to WaitForAppCenterBuild.

Reply:
I agree that further breakdown would improve the code quality and that can be postponed, but I'd still rather not leave the function name as being actively misleading. I suggested some changes.  Have a look when you get a chance.